### ISSUE TITLE
Incorrect reference flags for `x++` and `x--`

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#3326]
```js
let x;
x += 1;
x++;
++x;
x--;
--x;
```

`x += 1` gets correctly flagged as `Read | Write`. But all the rest (`x++` etc) are flagged as just `Write`. This appears to be incorrect.

[playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICkgICAgICAgIC2mcpqppRb4Fw6K%2FlQ6ebCAzKgaQCmbQvUeyG5MqU9%2Fdu%2Ff3EPr4A%3D)

**Comment by Dunqing** [comment]
This seems to be intentional

https://github.com/oxc-project/oxc/blob/e4b3a3c06a9d30a1fb60b1758bc9176377c48d58/crates/oxc_semantic/src/lib.rs#L273-L277

Same behavior in `no_unused_private_class_members` rule
https://github.com/oxc-project/oxc/blob/e4b3a3c06a9d30a1fb60b1758bc9176377c48d58/crates/oxc_linter/src/rules/eslint/no_unused_private_class_members.rs#L331-L337

But I think `x += 1` should be flagged as `Write` because `x` is never used for reading. I believe this variable can be safely removed by the Minifier/Compressor.

Does this make sense?

**Comment by overlookmotel** [comment]
Hmm. I'm not clear what these flags are meant to represent now!

These are all (broadly) equivalent:

```js
x = x + 1; // Read + Write
x += 1; // Read | Write
x++; // Write
++x; // Write
```

This also seems to be inconsistent - it flags `x` as `Read | Write` where it could be just `Write`:

```js
let x = 1;
let y = x = 2;
```

[playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICagICAgICAgIC2mcpqppTW%2BJ2DN6Fhf%2FaAzltG8wsRc39%2F7fiAgA%3D%3D)

And something strange going on here - the addition of `console.log` makes `x` `Read | Write`:

```js
console.log;
let x = 1;
x++;
```

[playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICggICAgICAgICxG0qZRWtpV52IeeJbVn4naXhvLZiVERX4nm00SA6ymugfDPN%2Ff8%2BEgIA%3D)

**Comment by Dunqing** [comment]
> Hmm. I'm not clear what these flags are meant to represent now!
> These are all (broadly) equivalent:
> 
> ```js
> x = x + 1; // Read + Write
> x += 1; // Read | Write
> x++; // Write
> ++x; // Write
> ```

The `x = x  + 1` is not equivalent. This assignment expression has two references. `x + 1` will be flagged `Read` and `x =` will be flagged `Write`
> 
> NB: Also depends on what `x` is. If it's an object with a `valueOf` method, the any of the above can have side effects.
> 
Yes! I forgot the `valueOf` method.

> This also seems to be inconsistent - it flags `x` as `Read | Write` where it could be just `Write`:
> 
> ```js
> let x = 1;
> let y = x = 2;
> ```
> 
> [playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICagICAgICAgIC2mcpqppTW%2BJ2DN6Fhf%2FaAzltG8wsRc39%2F7fiAgA%3D%3D)

This case is correct. `x = 2` should be flagged as `Write` and `y = x` should be flagged as `Read`  

> And something strange going on here - the addition of `console.log` makes `x` `Read | Write`:
> 
> ```js
> console.log;
> let x = 1;
> x++;
> ```
> 
> [playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICggICAgICAgICxG0qZRWtpV52IeeJbVn4naXhvLZiVERX4nm00SA6ymugfDPN%2Ff8%2BEgIA%3D)


This is a bug! I will fix it.

**Comment by magic-akari** [comment]
> This case is correct. `x = 2` should be flagged as `Write` and `y = x` should be flagged as `Read`
> 
> >

Here, x should be marked as `Write`, and y should also be marked as `Write`.
x will not be `Read`; `x = 2` returns the return value of the assignment expression, not the value read from x.

Please refer to:
https://x.com/magic_akari/status/1713914879449661734
